services/storage/tests/unit/test_dsm_dsmcleaner.py

```diff
@@ -88,7 +88,7 @@ async def test_clean_expired_uploads_deletes_expired_pending_uploads(
     """In this test we create valid upload links and check that once
     expired they get properly deleted"""
     await simcore_s3_dsm.create_file_upload_links(
-        user_id, simcore_file_id, link_type, file_size
+        user_id, simcore_file_id, link_type, file_size, False
```

Comment:
here I do not see any test where you have a directory defined?

Reply:
Yes, found it here: `services/storage/tests/unit/test_handlers_files.py::test_is_directory_link_forces_link_type_and_size`

Reply:
no, this test module tests the dsm_cleaner background task.
And in the current implementation there is not a single test checking that the cleaner does things correctly when a folder is defined.
What happens if:
1. you upload your directory and all goes well. the dsm cleaner should not remove the uploads (which I guess is implicitely tested in the test your refer to)
2. you upload your directory and the uploader is killed or loses network access. Who cleans the mess in S3 and potential multipart uploads that are dangling? Reminder: the multipart uploads are not free of charge, they cost. This test as far as I can tell is missing.

Reply:
The tests are now modified to deal with these cases.
1. this is now properly covered and the test was extended
2. the test for this now also takes into considerations directories, and if the links expire, they also get cleaned up.